buildtest/tools/build.py

```diff
@@ -72,9 +72,23 @@ def func_build_subcmd(args):
     print("{:_<80}".format(""))
     if args.config:
 
-        file = os.path.join(TESTCONFIG_ROOT, args.config)
+        # First try, the user can provide a full path
+        config_file = args.config
 
-        singlesource_test = SingleSource(file, args.collection, args.module_collection)
+        # Second try, the path can be relative to the TESTCONFIG_ROOT
+        if not os.path.exists(config_file):
+            config_file = os.path.join(TESTCONFIG_ROOT, config)
+
+        if not os.path.exists(config_file):
```

Comment:
@shahzebsiddiqui - 

 - the first check is if there is nothing provided, that means we use the entire test config root.
 - the second if else statement is essentially the same as doing isfile first, but it's a more concise way to write out the code. The way I implemented it results in the same result, but we don't need to call the os_walk command twice.

We can't technically check if it's a file or directory _first_ because it could be None, and that would result in an error. 

And yep, we are looping through the files and (for now) assuming single source. In the future this will be derive from the config files themselves.